Fetch previous messages and message date time fixes / improvements

BitPhoenix opened the pull request:
- Added dayjs to project (great library for working with dates / times)
- Now only showing h:mm PM/AM format in message component (in anticipation of having date separators like slack does (see rs_todo))
- Added fetchPreviousMessages function to chat component which will fetch the 25 messages most recent previous messages (works by sending over the sent_dt field of the oldest available message to the server to use in query)
- Right now this function is called on a button click but Timmy is going to have it execute on scroll up
- Created dateTimeService for getting the current date time as a string and for converting a date time string to a datetime (using dayjs instances)
- Updated message-service fetchChannelMessages and fetchPrivateMessages to accept optional parameter beforeDateTime. If provided, it will be sent over to the server in the request body and the server will use it in the query to fetch 25 most recent messages before that datetime

luiszaval commented on src/components/Message/Message.js:
```diff
@@ -1,7 +1,9 @@
+import { services } from "../../context";
 
 function Message(props) {
+    const sentDt = services.dateTimeService.dt(props.sent_dt).format("h:mm A")
```
could maybe do services.dateTimeService.formatForDisplay(props.send_dt)
& create the formatForDisplay (or whatever function name) function in dateTimeService since .format() is a method of dayjs. Not necessary but would keep all the dayjs functionality in dateTimeService

Could also destructure send_dt 
```
function Message({send_dt}) {
```

BitPhoenix replied:
Will update the sent_dt destructuring ✔️!

I can't quite do services.dateTimeService.formatForDisplay(props.send_dt) since we would want the flexibility to format date times for display in different ways depending on where it's being used, but I like like the idea of keeping dayjs functionality in the dateTimeService ✔️. 

I'll create dateTimeService.format(dateTime, format) where format is a string representing the format (retains flexibility to format a datetime in different ways throughout the app )

BitPhoenix replied:
Note: ended up naming this dateTimeService.str(dateTime, formatStr)

luiszaval commented on src/services/chat-service.js:
```diff
@@ -1,17 +1,19 @@
     // destination parameter can either be a username or channel name
     const prepareMessage = (type, content, sender, destination, orgName) => {
-        const dateOptions = { timeZone: "UTC", year: "numeric", month: "2-digit", day: "2-digit", hour: "numeric", minute: "numeric", hour12: true };
+        const date = new Date();
+        const sentDt = dateTimeService.getDtStr();
```
not 100% sure but looks like line 11 is being handled by line 12 when we call dayjs() in datetime-service

BitPhoenix replied:
Nice catch! This was accidentally left over from previous way of formatting date, will delete

luiszaval commented on src/services/chat-service.js:
```diff
@@ -1,17 +1,19 @@
 import { Subject } from "rxjs";
+const dayjs = require("dayjs");
```
was looking through the docs but did not see if this is here in order to maintain formatting? Else it looks like all dayjs functionality is handled in datetime-service and this can be deleted

BitPhoenix replied:
Yep also good catch, was accidentally left over from my initial implementation before I extracted into datetime-service. Will remove

luiszaval reviewed (changes requested):
Looks good!

Just a couple of things that may just come down to my lack of knowledge of new package

BitPhoenix commented:
Added new commit based on PR comments:
- removed FORMAT from datetime-service
- removed dateTimeService.getDtStr()
- added MESSAGE_DT_FORMAT to chat-service
- added DISPLAY_MESSAGE_DT_FORMAT to message component and getting RAW_MESSAGE_DT_FORMAT from services.chatService.MESSAGE_DT_FORMAT
- now destructuring props object in Message component
- Removed unnecessary dayjs import in chat-service
- Added dateTimeService.now() to return dateTime
- Added dateTimeService.str() to convert a dateTime to a string
